apps/checker/cmd/main.go

```diff
@@ -93,12 +103,58 @@ func main() {
 				return fmt.Errorf("unable to ping: %w", err)
 			}
 			statusCode := statusCode(res.StatusCode)
+
+			var isSuccessfull bool = true
+			if len(req.RawAssertions) > 0 {
+				for _, a := range req.RawAssertions {
+					var assert request.Assertion
+					err = json.Unmarshal(a, &assert)
+					if err != nil {
+						// handle error
+						return fmt.Errorf("unable to unmarshal assertion: %w", err)
+
+					}
+					switch assert.AssertionType {
+					case request.AssertionHeaders:
+						var target assertions.HeaderTarget
+						if err := json.Unmarshal(a, &target); err != nil {
+							return fmt.Errorf("unable to unmarshal IntTarget: %w", err)
+						}
+						isSuccessfull = isSuccessfull && target.HeaderEvaluate(res.Headers)
+
+						fmt.Println("assertion type", assert.AssertionType)
+					case request.AssertionTextBody:
+						fmt.Println("assertion type", assert.AssertionType)
+					case request.AssertionStatus:
+						var target assertions.StatusTarget
+						if err := json.Unmarshal(a, &target); err != nil {
+							return fmt.Errorf("unable to unmarshal IntTarget: %w", err)
+						}
+						isSuccessfull = isSuccessfull && target.StatusEvaluate(int64(res.StatusCode))
+					case request.AssertionJsonBody:
+						fmt.Println("assertion type", assert.AssertionType)
+					default:
+						fmt.Println("⚠️ Not Handled assertion type", assert.AssertionType)
+					}
+				}
+			} else {
+				isSuccessfull = statusCode.IsSuccessful()
+			}
+
 			// let's retry at least once if the status code is not successful.
-			if !statusCode.IsSuccessful() && called < 2 {
+			if !isSuccessfull && called < 2 {
```

Comment:
Should we **only** consider retry when status code is wrong, or the whole assertions?

Reply:
Whole assertion

To avoid false positive